swim/memberlist.go

```diff
@@ -291,10 +294,12 @@ func (m *memberlist) MakeFaulty(address string, incarnation int64) []Change {
 }
 
 func (m *memberlist) SetLocalStatus(status string) {
+	var before Member
 	m.members.Lock()
+	before = *m.local // make a copy of the before state
 	m.local.Status = status
 	m.members.Unlock()
-	m.postLocalUpdate()
+	m.postLocalUpdate(before)
```

Comment:
As discussed yesterday, It might be easier to refactor the postLocalUpdate to allow a modifier `func` to be passed in. Move the locking there, and let postLocalUpdate keep track of the before and after state.